### PR TITLE
Add User{primaryEmail} graphQL resolver

### PR DESCRIPTION
**Pull request by cbart**
Part of https://github.com/sourcegraph/sourcegraph/issues/49051

## Test plan

Table test checking reasonable corner cases:

- No emails - returns `null`
- One primary email and another non-primary - returns the primary email
- No email marked as primary - returns `null`

### CODE REVIEW

File: cmd/frontend/graphqlbackend/user_emails.go [hunk 1]
```diff
@@ -43,6 +43,32 @@ func (r *UserResolver) Emails(ctx context.Context) ([]*userEmailResolver, error)
...
 
+func (r *UserResolver) PrimaryEmail(ctx context.Context) (*userEmailResolver, error) {
+	// 🚨 SECURITY: Only the authenticated user and site admins can list user's
+	// emails on Sourcegraph.com.
+	if envvar.SourcegraphDotComMode() {
+		if err := auth.CheckSiteAdminOrSameUser(ctx, r.db, r.user.ID); err != nil {
+			return nil, err
+		}
+	}
+	ms, err := r.db.UserEmails().ListByUser(ctx, database.UserEmailsListOptions{
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/user_emails.go:54]: Duh that there's no method to get this directly. `GetPrimaryEmail` should probably just return `*UserEmail` :(

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -6356,6 +6356,11 @@ type User implements Node & SettingsSubject & Namespace {
     """
     emails: [UserEmail!]!
     """
+    The user's primary email address (if any).
```
Comment (limitedmage) [inline cmd/frontend/graphqlbackend/schema.graphql:6359]:
```suggestion
    The user's verified primary email address (if any).
```

**Review by limitedmage — COMMENTED** [review]
I feel that only *verified* emails should be returned here. Unverified emails should not be shown to other users.

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 2 above)
Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql:6359]: I wonder if there should even be non-verified primary emails at all 🤔 

-- 
`.Emails` returns all results, too, though so maybe this is a display concern? No strong opinion here either way, personally.

Comment (limitedmage) [reply cmd/frontend/graphqlbackend/schema.graphql:6359]: Hmm yeah don't feel super strong about this, just want to avoid users spamming non-verified addresses. I guess this is fine since it's not public (just on customer instances).

Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql:6359]: I wonder if non-verified should be returned _anywhere_ except in the user profile page `emails(includeUnverified: true)` in the GQL API.

File: cmd/frontend/graphqlbackend/user_emails.go (hunk 1 above)
Comment (cbart) [reply cmd/frontend/graphqlbackend/user_emails.go:54]: I don't think I understand.

- Do you mean the result of the graphQL resolver should be different?
- Or do you mean there should be a database op to return the primary email?

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 2 above)
Comment (cbart) [reply cmd/frontend/graphqlbackend/schema.graphql:6363]: Let's first make it return only verified email (I agree with the concern). Then we can add backwards-compatible parameter `includeUnverified: Boolean = false`
